python/plugins/processing/algs/qgis/CheckValidity.py

```diff
@@ -0,0 +1,182 @@
+# -*- coding: utf-8 -*-
+
+"""
+***************************************************************************
+    CheckValidity.py
+    ---------------------
+    Date                 : May 2015
+    Copyright            : (C) 2015 by Arnaud Morvan
+    Email                : arnaud dot morvan at camptocamp dot com
+***************************************************************************
+*                                                                         *
+*   This program is free software; you can redistribute it and/or modify  *
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+__author__ = 'Arnaud Morvan'
+__date__ = 'May 2015'
+__copyright__ = '(C) 2015, Arnaud Morvan'
+
+# This will get replaced with a git SHA1 when you do a git archive323
+
+__revision__ = '$Format:%H$'
+
+from PyQt4 import QtCore
+from qgis.core import QGis, QgsGeometry, QgsFeature, QgsField
+from processing.core.GeoAlgorithm import GeoAlgorithm
+from processing.core.parameters import ParameterVector
+from processing.core.parameters import ParameterSelection
+from processing.core.outputs import OutputVector
+from processing.tools import dataobjects, vector
+
+settings_method_key = "/qgis/digitizing/validate_geometries"
+
+
+class CheckValidity(GeoAlgorithm):
+
+    INPUT_LAYER = 'INPUT_LAYER'
+    METHOD = 'METHOD'
+    VALID_OUTPUT = 'VALID_OUTPUT'
+    INVALID_OUTPUT = 'INVALID_OUTPUT'
+    ERROR_OUTPUT = 'ERROR_OUTPUT'
+
+    METHODS = ['The one selected in digitizing settings',
```

Comment:
I think this is a good facility for the user, if he have selected a method in the settings, he won't have to select the method on each use.
I've heard many times this week, at the hackfest, users complain about having to select many times the same values in the processing dialogs.